sklearn/decomposition/tests/test_dict_learning.py

```diff
@@ -28,7 +28,12 @@
 def test_dict_learning_shapes():
     n_components = 5
     dico = DictionaryLearning(n_components, random_state=0).fit(X)
-    assert_true(dico.components_.shape == (n_components, n_features))
+    assert_equal(dico.components_.shape, (n_components, n_features))
+
+    n_components = 1
```

Comment:
@GaelVaroquaux this is the regression test for the SparseEncode change. I can add a more direct test on SparseEncode, though.